vitality/__init__.py

```diff
@@ -98,6 +99,34 @@ def profile():
         if not g.user:
             return redirect(url_for('login'))
         return render_template("profile.html")
+    
+    @app.route('/usersettings', methods=["GET","POST"])
+    def usersettings():
+        logger.info('Rendering User Settings')
+
+        if not g.user:
+            return redirect(url_for('login'))
+
+        if request.method == 'POST':
+            username = request.form['username']
+            password = request.form['password']
+            firstname = request.form['firstname']
+            lastname = request.form['lastname']
+            re_password = request.form['repassword']
+            location = request.form['location']
+            phone = request.form['phone']
+
+            if username and password == re_password:
+                g.user.username = username
+                g.user.password = password
+                g.user.firstname = firstname
+                g.user.lastname = lastname
+                g.user.location = location
+                g.user.phone = phone
```

Comment:
This is good, but I would rather create a new `User(...)` object and just assign `g.user = newUserObject` instead.